service/systemd/service_test.go

```diff
@@ -50,21 +49,21 @@ var listCmdArg = exec.RunParams{
 	Commands: `/bin/systemctl list-unit-files --no-legend --no-page -l -t service | grep -o -P '^\w[\S]*(?=\.service)'`,
 }
 
+var errFailure = errors.New("you-failed")
+
 type initSystemSuite struct {
 	coretesting.BaseSuite
 
 	dataDir string
 	ch      chan string
-	stub    *testing.Stub
-	conn    *systemd.StubDbusAPI
-	fops    *systemd.StubFileOps
-	exec    *systemd.StubExec
+	dBus    *MockDBusAPI
+	fops    *systemd.MockShimFileOps
```

Comment:
I guess I'm still not clear on how we get rid of these shims.

Reply:
I was unclear in my description. 

We would still need the same interfaces (maybe one all for side effects on the CLI/file-system), but they would have real implementers passed as dependencies instead of only existing for mocks that are patched over existing functions.